string text
float32 duration
float32 offset
float64 timestamp